scipy/stats/_qmc.py

```diff
@@ -1266,6 +1266,159 @@ def _random_oa_lhs(self, n: IntNumber = 4) -> np.ndarray:
+
+    The Hammersley sequence is the Halton sequence in dimensions :math:`n-1`
+    with a linear sweep in the last dimension.
+
+    Parameters
+    ----------
+    d : int
+        Dimension of the parameter space.
+    scramble : bool, optional
+        If True, use Owen scrambling for the Halton sequence and subtract
+        random offsets from the last dimension. Otherwise no scrambling is
+        done. Default is True.
+    optimization : {None, "random-cd", "lloyd"}, optional
+        Whether to use an optimization scheme to improve the quality after
+        sampling. Note that this is a post-processing step that does not
+        guarantee that all properties of the sample will be conserved.
+        Default is None.
+
+        * ``random-cd``: random permutations of coordinates to lower the
+          centered discrepancy. The best sample based on the centered
+          discrepancy is constantly updated. Centered discrepancy-based
+          sampling shows better space-filling robustness toward 2D and 3D
+          subprojections compared to using other discrepancy measures.
+        * ``lloyd``: Perturb samples using a modified Lloyd-Max algorithm.
+          The process converges to equally spaced samples.
+
+        .. versionadded:: 1.10.0
+    seed : {None, int, `numpy.random.Generator`}, optional
+        If `seed` is an int or None, a new `numpy.random.Generator` is
+        created using ``np.random.default_rng(seed)``.
+        If `seed` is already a ``Generator`` instance, then the provided
+        instance is used.
+
+    Notes
+    -----
+    The underlying Halton sequence has severe striping artifacts for even
+    modestly large dimensions. These can be ameliorated by scrambling.
+    Scrambling also supports replication-based error estimates and extends
+    applicabiltiy to unbounded integrands.
+
+    References
+    ----------
+    .. [1] Halton, "On the efficiency of certain quasi-random sequences of
+       points in evaluating multi-dimensional integrals", Numerische
+       Mathematik, 1960.
+    .. [2] A. B. Owen. "A randomized Halton algorithm in R",
+       :arxiv:`1706.02808`, 2017.
+    .. [3] "Low-Discrepancy sequence - Hammersley set". Wikipedia.
+       https://en.wikipedia.org/wiki/Low-discrepancy_sequence#Hammersley_set
+
+    Examples
+    --------
+    Generate samples from a low discrepancy sequence of Hammersley.
+
+    >>> from scipy.stats import qmc
+    >>> sampler = qmc.Hammersley(d=2, scramble=False)
+    >>> sample = sampler.random(n=5)
+    >>> sample
+    array([[0.   , 0.2  ],
+           [0.5  , 0.4  ],
+           [0.25 , 0.6  ],
+           [0.75 , 0.8  ],
+           [0.125, 1.   ]])
+
+    Compute the quality of the sample using the discrepancy criterion.
+
+    >>> qmc.discrepancy(sample)
+    0.0595017361111112
+
+    `random` may be called again, but because of the way the Hammersley
+    sequence is defined, the last column will still be based on linear sweep.
```

Comment:
Interesting, I will read more on that point. I am afraid about some clustering there. LHS for instance is a one shot method (our implementation at least, there are fancy methods to continue a sample if you are interested 😉), so I would not mind doing the same here.